pkg/client/bucket/bucket.go

```diff
@@ -85,17 +91,32 @@ func (c Client) upsert(ctx context.Context, bucketName string, data []byte) (Res
 		log.WithCtxFields(ctx).Debug("Created bucket with bucketName %q", bucketName)
 		return r, nil
 	}
+	log.WithCtxFields(ctx).WithFields(field.Error(err)).Debug("Failed to create new bucket with bucketName %q. Trying to update existing bucket. Error: %s", bucketName, err)
 
-	log.WithCtxFields(ctx).WithFields(field.Error(err)).Debug("Failed to create new object with bucketName %q. Trying to update existing object. Error: %s", bucketName, err)
+	for i := 0; i < retryStrategy.MaxRetries; i++ {
+		log.WithCtxFields(ctx).Debug("Trying to update bucket with bucketName %q (%d/%d retries)", bucketName, i+1, retryStrategy.MaxRetries)
 
-	b, err := c.Get(ctx, bucketName)
-	if err != nil {
-		return Response{}, fmt.Errorf("failed to get object with bucketName %q: %w", bucketName, err)
+		var b Response
+		b, err = c.Get(ctx, bucketName)
+		if err != nil {
+			return Response{}, fmt.Errorf("failed to get bucket with bucketName %q: %w", bucketName, err)
+		}
+
+		r, err = c.update(ctx, b, data)
+		if err == nil {
+			log.WithCtxFields(ctx).Debug("Updated bucket with bucketName %q", bucketName)
+			return r, nil
+		}
+
+		var respErr rest.RespError
+		if !(errors.As(err, &respErr) && respErr.StatusCode == http.StatusConflict) {
```

Comment:
nit: I wonder if this would be easier to read if it wasn't inverted but an "early exit" type sleep + continue if there's a Conflict HTTP status error. And the err == nil block was the normal path instead of a condition

Reply:
I'd stay we can try it in the shared lib :)